traffic_ops/client/user.go

```diff
@@ -36,8 +37,26 @@ func (to *Session) Users() ([]tc.User, error) {
 
 // GetUsers returns all users accessible from current user
 func (to *Session) GetUsers() ([]tc.User, ReqInf, error) {
-	route := apiBase + "/users.json"
-	resp, remoteAddr, err := to.request("GET", route, nil)
+	route := apiBase + "/users"
+	resp, remoteAddr, err := to.request(http.MethodGet, route, nil)
+	reqInf := ReqInf{CacheHitStatus: CacheHitStatusMiss, RemoteAddr: remoteAddr}
+	if err != nil {
+		return nil, reqInf, err
+	}
+	defer resp.Body.Close()
+
+	var data tc.UsersResponse
+	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+		return nil, reqInf, err
+	}
+
+	return data.Response, reqInf, nil
```

Comment:
Why not just return the error, instead of using an `if` to explicitly return it if it's not `nil` and return `nil` when it's `nil`?

Reply:
i don't follow. this seems right to me. if there's an error, return nil for the slice of users and return the error. oh, are you saying on this line:

`return data.Response, reqInf, nil`

change it to 

`return data.Response, reqInf, err`

Reply:
oh, i see what's your saying. lots of extra code.

Reply:
yes exactly

Reply:
well `GetUsers` i didn't write so hesitant to rewrite it. although it is under test so i guess i could.

Reply:
well you don't need to rewrite `GetUsers` if you don't want to. Git diffs are weird, I put this in the part it said you changed but it looks like what you actually did was remove `.json` and add the `GetUsersByRole` method, so really it just ought to be changed in `GetUsersByRole`. Though I'd consider the same change to `GetUsers` an improvement and encourage you to do that too, if you want.